test/app.test.ts

```diff
@@ -1,10 +1,18 @@
+import Koa from 'koa';
 import parseLinkHeader from 'parse-link-header';
 import request from 'supertest';
-import app from '../src/app';
+import InMemoryArticles from '../src/adaptors/in-memory-articles';
+import createApp from '../src/app';
 
 const trim = (value: string): string => value.trim();
 const parseHeader = (header: string): Array<string> => header.split(',').map(trim);
 
+let app: Koa;
+
+beforeEach((): void => {
+  app = createApp(new InMemoryArticles());
+});
```

Comment:
I understand no tear down is needed because this `app` instance doesn't listen on a port but is only used through in-process tests. Therefore it will just be garbage collected. The new instance of `InMemoryArticles` also ensures no state is carry over between tests.